parachain/primitives/router/src/inbound/mod.rs

```diff
@@ -87,15 +91,33 @@ pub trait ConvertMessage {
 
 pub type CallIndex = [u8; 2];
 
-impl<CreateAssetCall> ConvertMessage for MessageToXcm<CreateAssetCall>
+pub trait XcmFeeGetter {
+	// Measures the maximum amount of gas a command will require
+	fn fee(command: &Command) -> u128;
+}
+
+/// A meter that assigns a constant amount of gas for the execution of a command
+pub struct ConstantFeeForInboundMessage;
+
+impl XcmFeeGetter for ConstantFeeForInboundMessage {
+	fn fee(command: &Command) -> u128 {
+		match command {
+			Command::RegisterToken { .. } => 2_000_000_000,
+			Command::SendToken { .. } => 1_000_000_000,
+		}
+	}
+}
+
+impl<CreateAssetCall, FeeGetter> ConvertMessage for MessageToXcm<CreateAssetCall, FeeGetter>
 where
 	CreateAssetCall: Get<CallIndex>,
+	FeeGetter: XcmFeeGetter,
 {
 	fn convert(message: VersionedMessage) -> Result<Xcm<()>, ConvertMessageError> {
 		match message {
-			VersionedMessage::V1(MessageV1 { chain_id, message }) => {
+			VersionedMessage::V1(MessageV1 { chain_id, message: command }) => {
 				let network = Ethereum { chain_id };
-				let buy_execution_fee_amount = 2_000_000_000;
+				let buy_execution_fee_amount = FeeGetter::fee(&command);
```

Comment:
I am not too sure about this design. For one, we may need to construct an xcm message that buys execution fees multiple times, for example, on AssetHub, and then the final destination parachain.

So I think we should start simple and rather rely on the following generic parameters:

```rust
CreateAssetExecutionFee: Get<u128>
SendTokenExecutionFee: Get<u128>
```

Then if we need more complexity, we can revisit as necessary

Reply:
Yean make sense. Could be some double map storage like `(Multilocation,Command)->Fee` but let's make it simple now. Some revamp in
https://github.com/Snowfork/snowbridge/pull/970/commits/31e3019f3c9d1b5ed2951e2eeb2460824cd43ad4